File: Games/BombermanGame.Shared/Interfaces/IGameInfo.cs
using Games.BombermanGame.Shared.Enums;

namespace Games.BombermanGame.Shared.Interfaces
{
    /// <summary>
    /// Information about the game stage
    /// </summary>
    public interface IGameInfo
    {
        FieldItemEnum[][] Field { get; }
        int FieldHeight { get; }
        int FieldWidth { get; }
        
        IPlayerInfo[] PlayersInfos { get; }
    }
}

File: BombermanClient_C_PlusPlus/Games/BombermanGame.Shared/Interfaces/IGameInfo.cpp
//
// Created by Andrey on 02.12.2023.
//

#include "IGameInfo.h"

namespace Games_BombermanGame_Shared_Interfaces {
    FieldItemEnum **IGameInfo::getField() const {
        return _field;
    }

    int IGameInfo::getFieldHeight() const {
        return _fieldHeight;
    }

    int IGameInfo::getFieldWidth() const {
        return _fieldWidth;
    }

    IPlayerInfo *IGameInfo::getPlayersInfos() const {
        return _playersInfos;
    }
} // Games_BombermanGame_Shared_Interfaces